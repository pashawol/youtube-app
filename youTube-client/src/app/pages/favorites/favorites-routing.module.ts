import { NgModule } from "@angular/core"
import { RouterModule, Routes } from "@angular/router"

import { FavoritesContainerComponent } from "./containers/favorites-container/favorites-container.component"

const routes: Routes = [
    {
        path: "",
        component: FavoritesContainerComponent
    }
]

@NgModule({
    imports: [RouterModule.forChild(routes)],
    exports: [RouterModule]
})
export class FavoritesRoutingModule {}
