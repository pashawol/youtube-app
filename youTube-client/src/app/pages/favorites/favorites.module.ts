import { CommonModule } from "@angular/common"
import { NgModule } from "@angular/core"

import { FavoritesRoutingModule } from "./favorites-routing.module"

@NgModule({
    declarations: [],
    imports: [CommonModule, FavoritesRoutingModule]
})
export class FavoritesModule {}
