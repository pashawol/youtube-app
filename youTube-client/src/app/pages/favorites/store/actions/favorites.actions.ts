import { Item } from "@app/shared/models/response.model"
import { createAction, props } from "@ngrx/store"

export const loadFavorites = createAction("[Favorites] Load Favorites")
export const loadFavoritesSuccess = createAction("[Favorites] Load Favorites Success", props<{ favorites: string[] }>())

export const toggleFavorite = createAction("[Favorite] Toggle Favorite", props<{ id: string }>())

export const loadFavoritesData = createAction("[Favorites] Load Favorites Data")
export const loadFavoritesDataSuccess = createAction(
    "[Favorites] Load Favorites Data Success",
    props<{ favoritesData: Item[] }>()
)
export const loadFavoritesDataFailure = createAction(
    "[Favorites] Load Favorites Data Failure",
    props<{ error: string | null }>()
)
