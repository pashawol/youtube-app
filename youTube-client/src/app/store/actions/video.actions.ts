import { createAction, props } from "@ngrx/store"
import { Item } from "@shared/models/response.model"

import { VideoState } from "../videos.state"

export const getVideoById = createAction("[Video] Get Video By Id", props<{ id: string }>())
export const getVideoByIdSuccess = createAction("[Video] Get Video By Id Success", props<{ video: Item }>())
export const getVideoByIdFailure = createAction("[Video] Get Video By Id Failure", props<{ error: string | null }>())

export const loadVideos = createAction("[Video] Load Videos")
export const loadVideosSuccess = createAction("[Video] Load Videos Success", props<{ videos: Item[] }>())
export const loadVideosFailure = createAction("[Video] Load Videos Failure", props<{ error: string | null }>())

export const createVideo = createAction("[Video] Create Video", props<{ video: Item }>())
export const createVideoFailure = createAction("[Video] Create Video Failure", props<{ error: string | null }>())

export const deleteVideo = createAction("[Video] Delete Video", props<{ id: string }>())
