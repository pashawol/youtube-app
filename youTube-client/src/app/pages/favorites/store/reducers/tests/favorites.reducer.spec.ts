import { Item } from "@app/shared/models/response.model"

import * as FavoritesActions from "../../actions/favorites.actions"
import { initialState } from "../../favorites.state"
import { favoritesReducer } from "../favorites.reducer"

const newVideo: Item = {
    kind: "kind",
    etag: "etag",
    id: "id",
    snippet: {
        publishedAt: "publishedAt",
        channelId: "channelId",
        title: "title",
        description: "description",
        thumbnails: {
            default: {
                url: "url",
                width: 0,
                height: 0
            },
            medium: {
                url: "url",
                width: 0,
                height: 0
            },
            high: {
                url: "url",
                width: 0,
                height: 0
            }
        },
        channelTitle: "channelTitle",
        tags: ["tags"],
        categoryId: "categoryId",
        liveBroadcastContent: "liveBroadcastContent",
        localized: {
            title: "title",
            description: "description"
        },
        defaultAudioLanguage: "defaultAudioLanguage",
        defaultLanguage: "defaultLanguage"
    },
    statistics: {
        viewCount: "viewCount",
        likeCount: "likeCount",
        dislikeCount: "dislikeCount",
        favoriteCount: "favoriteCount",
        commentCount: "commentCount"
    }
}
describe("favoritesReducer", () => {
    it("should return the initial state", () => {
        const state = favoritesReducer(undefined, { type: "@@INIT" })
        expect(state).toMatchSnapshot()
    })

    it("should handle toggleFavorite action", () => {
        const action = FavoritesActions.toggleFavorite({ id: "id" })
        const state = favoritesReducer(initialState, action)

        expect(state).toMatchSnapshot()
    })

    it("should handle loadFavoritesDataSuccess action", () => {
        const action = FavoritesActions.loadFavoritesDataSuccess({ favoritesData: [newVideo] })
        const state = favoritesReducer(initialState, action)

        expect(state).toMatchSnapshot()
    })
})
