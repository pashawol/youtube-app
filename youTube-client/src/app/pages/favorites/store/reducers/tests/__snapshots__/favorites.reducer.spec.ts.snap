// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoritesReducer should handle loadFavoritesDataFailure action 1`] = `
{
  "error": "error",
  "favoriteList": [
    "7PxxZWrlRTc",
    "Dh-k6EcwPmI",
  ],
  "favoritesData": [],
}
`;

exports[`favoritesReducer should handle loadFavoritesDataSuccess action 1`] = `
{
  "error": null,
  "favoriteList": [
    "7PxxZWrlRTc",
    "Dh-k6EcwPmI",
  ],
  "favoritesData": [
    {
      "etag": "etag",
      "id": "id",
      "kind": "kind",
      "snippet": {
        "categoryId": "categoryId",
        "channelId": "channelId",
        "channelTitle": "channelTitle",
        "defaultAudioLanguage": "defaultAudioLanguage",
        "defaultLanguage": "defaultLanguage",
        "description": "description",
        "liveBroadcastContent": "liveBroadcastContent",
        "localized": {
          "description": "description",
          "title": "title",
        },
        "publishedAt": "publishedAt",
        "tags": [
          "tags",
        ],
        "thumbnails": {
          "default": {
            "height": 0,
            "url": "url",
            "width": 0,
          },
          "high": {
            "height": 0,
            "url": "url",
            "width": 0,
          },
          "medium": {
            "height": 0,
            "url": "url",
            "width": 0,
          },
        },
        "title": "title",
      },
      "statistics": {
        "commentCount": "commentCount",
        "dislikeCount": "dislikeCount",
        "favoriteCount": "favoriteCount",
        "likeCount": "likeCount",
        "viewCount": "viewCount",
      },
    },
  ],
}
`;

exports[`favoritesReducer should handle toggleFavorite action 1`] = `
{
  "error": null,
  "favoriteList": [
    "7PxxZWrlRTc",
    "Dh-k6EcwPmI",
    "id",
  ],
  "favoritesData": [],
}
`;

exports[`favoritesReducer should return the initial state 1`] = `
{
  "error": null,
  "favoriteList": [
    "7PxxZWrlRTc",
    "Dh-k6EcwPmI",
  ],
  "favoritesData": [],
}
`;
